### PR TITLE
add some information in GetTableInfoBySQL

### LINKED ISSUE #590: schemacmp.Table.String() return incorrect `create table` SQL
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?

	In DM

	1. create table 
	  ```sql
	  create table precheck_optimistic_tb_1(a INT, b INT, primary key a(a));
	  ```
	2. get `tableInfo` from GetTableInfoBySQL
	3. encode `tableInfo` to `schemacmp.Table`
	4. call schemacmp.Table.String()
	5. get 
	  ```sql
	  CREATE TABLE `tbl`(`a` INT(11) NOT NULL, `b` INT(11), PRIMARY KEY USING  (`a`(0))) CHARSET UTF8MB4 COLLATE UTF8MB4_BIN
	  ```

2. What did you expect to see?
```
CREATE TABLE `tbl`(`a` INT(11) NOT NULL, `b` INT(11), PRIMARY KEY  (`a`)) CHARSET UTF8MB4 COLLATE UTF8MB4_BIN
```
3. What did you see instead?
```
CREATE TABLE `tbl`(`a` INT(11) NOT NULL, `b` INT(11), PRIMARY KEY USING  (`a`(0))) CHARSET UTF8MB4 COLLATE UTF8MB4_BIN
```
4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?

5. which tool are you using?

6. what versionof tool are you using (`pump -V` or `tidb-lightning -V` or `syncer -V`)?

### reason

https://github.com/pingcap/tidb-tools/blob/d9fdfa2f9040aab3fab7cd11774a82226f467fe7/pkg/dbutil/table.go#L59-L73

not assign `model.IndexInfo.Tp` and `model.IndexInfo.Column.Length`

### PR DESCRIPTION
**Pull request by okJiang**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb-tools/issues/590

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

### CODE REVIEW

**Comment by lance6716** [comment]
need an unit test

**Comment by okJiang** [comment]
after the issue updated, so need this pr add unit test yet? @lance6716 @GMHDBJD

**Comment by lance6716** [comment]
> after the issue updated, so need this pr add unit test yet? @lance6716 @GMHDBJD

I prefer an unit test to let reviewers easily and visually check the output

File: pkg/dbutil/table_test.go [hunk 1]
```diff
@@ -33,6 +34,7 @@ type testCase struct {
 	sql     string
 	columns []string
 	indexs  []string
+	colLen  [][]int
```
Comment (lance6716) [inline pkg/dbutil/table_test.go:38]: not this type of UT, I mean we should check the https://github.com/pingcap/tidb-tools/issues/590 can has expected SQL

**Review by lance6716 — APPROVED** [review]

**Comment by okJiang** [comment]
if no other questions, can you give me a merge to close this pr? @lance6716

**Pull request merged**